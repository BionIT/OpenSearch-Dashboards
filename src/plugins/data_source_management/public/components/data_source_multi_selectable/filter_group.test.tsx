/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { ShallowWrapper, shallow } from 'enzyme';
import React from 'react';
import { FilterGroup } from './filter_group';

describe('FilterGroup', () => {
  let component: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

  it('should render normally', () => {
    const setItems = jest.fn();
    component = shallow(
      <FilterGroup
        items={[{ id: '1', name: 'name1', checked: 'on', visible: true }]}
        setItems={(items) => setItems(items)}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
