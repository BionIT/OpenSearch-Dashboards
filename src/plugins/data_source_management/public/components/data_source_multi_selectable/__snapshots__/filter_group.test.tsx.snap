// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterGroup should render normally 1`] = `
<EuiPopover
  anchorPosition="downCenter"
  button={
    <React.Fragment>
      <EuiButtonEmpty
        className="euiHeaderLink"
        iconType="database"
        onClick={[Function]}
        size="s"
      >
        Data sources
      </EuiButtonEmpty>
      <EuiNotificationBadge
        color="subdued"
      >
        1
      </EuiNotificationBadge>
    </React.Fragment>
  }
  closePopover={[Function]}
  display="inlineBlock"
  hasArrow={true}
  id="popoverExampleMultiSelect"
  isOpen={false}
  ownFocus={true}
  panelPaddingSize="none"
>
  <EuiPopoverTitle
    paddingSize="s"
  >
    <EuiFieldSearch
      compressed={true}
      fullWidth={false}
      incremental={false}
      isClearable={true}
      isLoading={false}
      onSearch={[Function]}
    />
  </EuiPopoverTitle>
  <div
    className="ouiFilterSelect__items"
    style={
      Object {
        "maxHeight": 400,
        "overflow": "scroll",
      }
    }
  >
    <EuiFilterSelectItem
      checked="on"
      key="0"
      onClick={[Function]}
      showIcons={true}
      style={
        Object {
          "display": undefined,
          "paddingLeft": undefined,
        }
      }
    >
      name1
    </EuiFilterSelectItem>
  </div>
  <EuiPopoverFooter>
    <EuiButtonGroup
      idSelected="select_all"
      legend="All toptions selection toggle group"
      onChange={[Function]}
      options={
        Array [
          Object {
            "id": "select_all",
            "label": "Select all",
          },
          Object {
            "id": "deselect_all",
            "label": "Deselect all",
          },
        ]
      }
    />
  </EuiPopoverFooter>
</EuiPopover>
`;
